enigma/src/main/java/cuchaz/enigma/EnigmaProject.java

```diff
@@ -344,6 +344,11 @@ public static class ClassSource {
 		}
 
 		public void writeTo(Path path) throws IOException {
+			if (!Files.exists(path)) {
+				Files.createDirectories(path.getParent());
+				Files.createFile(path);
+			}
```

Comment:
```suggestion
			Files.createDirectories(path.getParent());
```

Reply:
i tested whether the Files writer creates file automatically and i found that the function creates file but not directories. the code above is best code in single class i think